sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +353,39 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task GetPulumiVersionAsync(CancellationToken cancellationToken = default)
```

Comment:
Why is `= default` needed?

Reply:
It's like that for every single other `Async` function in this file 🤷🏽

Reply:
We're doing it on the public APIs so that consumers can choose to provide or not provide a cancellation token - not everybody does and generally it is commonplace to have overloads for both. We may have issues in the future if we need to make changes to the parameters on those APIs, but cancellation token being the final optional parameter (in ordering) makes it relatively safe.

In this scenario with a private method that will only be called further down the method chain we can make the assumption that we will always have a cancellation token - because the method further up the chain that is calling this method already took in the optional. So here it is not necessary.

Reply:
Got it! FWIW it looks like [CreateStackAsync](https://github.com/pulumi/pulumi/blob/master/sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs#L517) is also a public method that does not have ` = default`

Reply:
On first glance that looks like an oversight, but if you look at the [abstract Workspace](https://github.com/pulumi/pulumi/blob/b187ce18c24b2229531f1e9100334e672976f451/sdk/dotnet/Pulumi.Automation/Workspace.cs#L185) that `LocalWorkspace` extends then you'll see the overloads that allow you to exclude a cancellation token.

Reply:
I believe that for `CreateStackAsync` and `SelectStackAsync` I needed to use concrete parameters and provide physical overloads (instead of optional parameters) because if the parameters were optional it was causing method signature collisions with the [static methods](https://github.com/pulumi/pulumi/blob/b187ce18c24b2229531f1e9100334e672976f451/sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs#L84).